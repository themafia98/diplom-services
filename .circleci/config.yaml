version: 2
jobs:
  build:
docker: # use the docker executor type; machine and macos executors are also supported
  - image: circleci/node:13.3.0 # the primary container, where your job's commands are run
steps:
  - checkout # check out the code in the project directory
  - run: echo "npm installing" # run the `echo` command
  - run: npm install
  - run: npm build
  test:
docker: # use the docker executor type; machine and macos executors are also supported
  - image: circleci/node:13.3.0 # the primary container, where your job's commands are run
steps:
  - checkout # check out the code in the project directory
  - run: echo "Testing Code" #
  - run: npm install
  - run: npm test
workflows:
  version: 2
  build_test_and_lint:
jobs:
  - build
  - test
