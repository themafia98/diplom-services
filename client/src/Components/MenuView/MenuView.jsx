import React from 'react';
import { menuViewType } from './types';
import { Menu, Layout, Icon } from 'antd';

import Logo from 'Components/Logo';

const { Sider } = Layout;
const { SubMenu } = Menu;

const MenuView = ({ collapsed, cbOnCollapse, items, cbMenuHandler, activeTabEUID, cbGoMain }) => {
  const renderMenu = (items) =>
    items
      .filter(({ PARENT_CODE = '' }) => PARENT_CODE === null)
      .map(({ EUID = '', ICON = '', VALUE = '' }) => {
        const children = items.filter(({ PARENT_CODE }) => EUID === PARENT_CODE);
        if (children.length > 0) {
          return (
            <SubMenu
              key={EUID}
              className="menuItem"
              title={
                <span>
                  {ICON ? <Icon type={ICON} /> : null}
                  <span>{VALUE}</span>
                </span>
              }
            >
              {children.map((child) => (
                <Menu.Item onClick={cbMenuHandler} key={child.EUID}>
                  {child.VALUE}
                </Menu.Item>
              ))}
            </SubMenu>
          );
        } else
          return (
            <Menu.Item className="menuItem" onClick={cbMenuHandler} key={EUID}>
              {ICON ? <Icon type={ICON} /> : null}
              <span>{VALUE}</span>
            </Menu.Item>
          );
      });

  return (
    <Sider collapsible collapsed={collapsed} onCollapse={cbOnCollapse}>
      <div className="logo" onClick={cbGoMain}>
        <Logo />
      </div>
      {items ? (
        <Menu
          selectedKeys={[activeTabEUID]}
          theme="dark"
          defaultSelectedKeys={[items[0].VALUE]}
          mode="inline"
        >
          {renderMenu(items)}
        </Menu>
      ) : null}
    </Sider>
  );
};

MenuView.defaultProps = {
  collapsed: false,
  cbOnCollapse: null,
  items: [],
  cbMenuHandler: null,
  activeTabEUID: '',
  cbGoMain: null,
};
MenuView.propTypes = menuViewType;

export default MenuView;
