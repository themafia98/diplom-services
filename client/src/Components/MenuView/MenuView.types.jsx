import PropTypes from 'prop-types';
const { bool, func, oneOfType, object, string, oneOf } = PropTypes;

export const menuViewType = {
  collapsed: bool.isRequired,
  cbOnCollapse: func.isRequired,
  items: oneOfType([PropTypes.arrayOf(object.isRequired).isRequired, oneOf([null])]),
  cbMenuHandler: func.isRequired,
  activeTabEUID: string.isRequired,
  cbGoMain: func.isRequired,
};
