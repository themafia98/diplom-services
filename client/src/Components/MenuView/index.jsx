import React from "react";
import confing from "../../config.json";
import _ from "lodash";
import { Menu, Layout, Icon } from "antd";

const { Sider } = Layout;
const { SubMenu } = Menu;

const MenuView = ({ collapsed, cbOnCollapse, items, cbMenuHandler, activeTabEUID, cbGoMain }) => {
    const renderMenu = items => {
        const parent = items.filter(parentItem => {
            return _.isNull(parentItem.PARENT_CODE);
        });

        return parent.map(item => {
            const children = items.filter(childParentItem => {
                return item.EUID === childParentItem.PARENT_CODE;
            });
            if (children.length > 0) {
                return (
                    <SubMenu
                        key={item.EUID}
                        title={
                            <span>
                                {item.ICON ? <Icon type={item.ICON} /> : null}
                                <span>{item.VALUE}</span>
                            </span>
                        }
                    >
                        {children.map(child => (
                            <Menu.Item onClick={cbMenuHandler} key={child.EUID}>
                                {child.VALUE}
                            </Menu.Item>
                        ))}
                    </SubMenu>
                );
            } else
                return (
                    <Menu.Item onClick={cbMenuHandler} key={item.EUID}>
                        {item.ICON ? <Icon type={item.ICON} /> : null}
                        <span>{item.VALUE}</span>
                    </Menu.Item>
                );
        });
    };

    return (
        <Sider collapsible collapsed={collapsed} onCollapse={cbOnCollapse}>
            <div className="logo" onClick={cbGoMain}>
                {/*<span>{confing.title}</span>*/}
                <img src={confing.icon} alt="logo" />
            </div>
            {items ? (
                <Menu selectedKeys={[activeTabEUID]} theme="dark" defaultSelectedKeys={[items[0].VALUE]} mode="inline">
                    {renderMenu(items)}
                </Menu>
            ) : null}
        </Sider>
    );
};

export default MenuView;
