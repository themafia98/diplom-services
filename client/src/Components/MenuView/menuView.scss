ul {
    &.ant-menu{
        &.ant-menu-sub, &.ant-menu-dark {
            li {
                &.ant-menu-item-selected {
                    background-color: unset;
                }
            }
        }
    }
}

.aside_menu {
    flex: 0 0 210px;
    max-width: 210px;
    min-width: 209px;
    width: 210px;
}