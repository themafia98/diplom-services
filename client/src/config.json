{
    "title": "Controll system",
    "icon": "/img/logo.svg",
    "firebase": {
        "session": "LOCAL",
        "lowConnection": "false"
    },
    "clientDB": {
        "name": "controllSystemDB",
        "version": 1
    },
    "forceUpdate": false,
    "msTimeoutLoading": 0,
    "tabsLimit": 15,
    "menu": [
        {
            "EUID": "mainModule",
            "VALUE": "Главное меню",
            "ICON": "home",
            "SIGN": "default",
            "PARENT_CODE": null
        },
        {
            "EUID": "cabinetModule",
            "VALUE": "Личный кабинет",
            "ICON": "user",
            "PARENT_CODE": null
        },
        {
            "EUID": "taskModule",
            "VALUE": "Задачи",
            "ICON": "file-done",
            "PARENT_CODE": null
        },
        {
            "EUID": "taskModule_all",
            "VALUE": "Все задачи",
            "ICON": null,
            "PARENT_CODE": "taskModule"
        },
        {
            "EUID": "taskModule_myTasks",
            "VALUE": "Мои задачи",
            "ICON": null,
            "PARENT_CODE": "taskModule"
        },
        {
            "EUID": "taskModule_сalendar",
            "VALUE": "Календарь задач",
            "ICON": null,
            "PARENT_CODE": "taskModule"
        },
        {
            "EUID": "taskModule_createTask",
            "VALUE": "Новая задача",
            "ICON": null,
            "SIGN": "freedom"
        },
        {
            "EUID": "customersModule",
            "VALUE": "Работа с клиентами",
            "ICON": "global",
            "PARENT_CODE": null
        },
        {
            "EUID": "customersModule_contacts",
            "VALUE": "Контакты",
            "ICON": null,
            "PARENT_CODE": "customersModule"
        },
        {
            "EUID": "customersModule_reception",
            "VALUE": "Обращения",
            "ICON": null,
            "PARENT_CODE": "customersModule"
        },
        {
            "EUID": "contactModule",
            "VALUE": "Связь",
            "ICON": "wechat",
            "PARENT_CODE": null
        },
        {
            "EUID": "contactModule_feedback",
            "VALUE": "Информация",
            "ICON": null,
            "PARENT_CODE": "contactModule"
        },
        {
            "EUID": "contactModule_chat",
            "VALUE": "Корпоративный чат",
            "ICON": null,
            "PARENT_CODE": "contactModule"
        },
        {
            "EUID": "statisticModule",
            "VALUE": "Статистика",
            "ICON": "line-chart",
            "PARENT_CODE": null
        },
        {
            "EUID": "settingsModule",
            "VALUE": "Настройки",
            "ICON": "build",
            "PARENT_CODE": null
        }
    ]
}
