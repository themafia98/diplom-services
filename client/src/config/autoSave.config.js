const autoSaveConfig = {
  interval: 10000,
  validKeysRouteForSave: [
    'path',
    'currentActionTab',
    'activeTabs',
    'routeDataActive',
    'routeData',
    'load',
    'partDataPath',
    'shouldUpdate',
  ],
};

export default autoSaveConfig;
