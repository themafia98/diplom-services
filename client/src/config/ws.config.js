const ws = {
  startUrl: '/',
  socketIO: {
    path: '/socket.io',
    transports: ['websocket'],
    secure: false,
  },
};

export default ws;
