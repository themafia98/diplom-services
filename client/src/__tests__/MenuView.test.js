import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import config from "../config.json";

import MenuView from "../Components/MenuView/index";

describe("<MenuView />", () => {
    test("Should MenuView render with itemsMenu", () => {
        const props = {
            collapsed: true,
            cbOnCollapse: () => {},
            items: config.menu,
            cbMenuHandler: () => {},
            activeTabEUID: "MainModule",
            cbGoMain: () => {},
        };
        const MenuViewWrapper = shallow(<MenuView {...props} />);
        expect(toJson(MenuViewWrapper)).toMatchSnapshot();

        MenuViewWrapper.find(".menuItem").forEach(node => {
            expect(node.hasClass("menuItem")).toEqual(true);
            expect(node.exists()).toEqual(true);
        });

        expect(MenuViewWrapper.find("Menu").prop("defaultSelectedKeys")).toEqual([props.items[0].VALUE]);

        expect(MenuViewWrapper.find("Sider").simulate("click", []));
        expect(toJson(MenuViewWrapper)).toMatchSnapshot();
    });
});
