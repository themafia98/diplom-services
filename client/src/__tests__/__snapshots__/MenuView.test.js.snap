// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuView /> Should MenuView render with itemsMenu 1`] = `
<Sider
  collapsed={true}
  collapsible={true}
  onCollapse={[Function]}
>
  <div
    className="logo"
    onClick={[Function]}
  >
    <Logo />
  </div>
  <Menu
    defaultSelectedKeys={
      Array [
        "Главное меню",
      ]
    }
    mode="inline"
    selectedKeys={
      Array [
        "MainModule",
      ]
    }
    theme="dark"
  >
    <MenuItem
      className="menuItem"
      key="mainModule"
      onClick={[Function]}
    >
      <Icon
        type="home"
      />
      <span>
        Главное меню
      </span>
    </MenuItem>
    <MenuItem
      className="menuItem"
      key="cabinetModule"
      onClick={[Function]}
    >
      <Icon
        type="user"
      />
      <span>
        Личный кабинет
      </span>
    </MenuItem>
    <SubMenu
      className="menuItem"
      key="taskModule"
      title={
        <span>
          <Icon
            type="file-done"
          />
          <span>
            Задачи
          </span>
        </span>
      }
    >
      <MenuItem
        key="taskModule_all"
        onClick={[Function]}
      >
        Все задачи
      </MenuItem>
      <MenuItem
        key="taskModule_myTasks"
        onClick={[Function]}
      >
        Мои задачи
      </MenuItem>
      <MenuItem
        key="taskModule_сalendar"
        onClick={[Function]}
      >
        Календарь задач
      </MenuItem>
    </SubMenu>
    <SubMenu
      className="menuItem"
      key="customersModule"
      title={
        <span>
          <Icon
            type="global"
          />
          <span>
            Работа с клиентами
          </span>
        </span>
      }
    >
      <MenuItem
        key="customersModule_contacts"
        onClick={[Function]}
      >
        Контакты
      </MenuItem>
      <MenuItem
        key="customersModule_reception"
        onClick={[Function]}
      >
        Обращения
      </MenuItem>
    </SubMenu>
    <SubMenu
      className="menuItem"
      key="contactModule"
      title={
        <span>
          <Icon
            type="wechat"
          />
          <span>
            Связь
          </span>
        </span>
      }
    >
      <MenuItem
        key="contactModule_feedback"
        onClick={[Function]}
      >
        Информация
      </MenuItem>
      <MenuItem
        key="contactModule_chat"
        onClick={[Function]}
      >
        Корпоративный чат
      </MenuItem>
    </SubMenu>
    <MenuItem
      className="menuItem"
      key="statisticModule"
      onClick={[Function]}
    >
      <Icon
        type="line-chart"
      />
      <span>
        Статистика
      </span>
    </MenuItem>
    <MenuItem
      className="menuItem"
      key="settingsModule"
      onClick={[Function]}
    >
      <Icon
        type="build"
      />
      <span>
        Настройки
      </span>
    </MenuItem>
  </Menu>
</Sider>
`;

exports[`<MenuView /> Should MenuView render with itemsMenu 2`] = `
<Sider
  collapsed={true}
  collapsible={true}
  onCollapse={[Function]}
>
  <div
    className="logo"
    onClick={[Function]}
  >
    <Logo />
  </div>
  <Menu
    defaultSelectedKeys={
      Array [
        "Главное меню",
      ]
    }
    mode="inline"
    selectedKeys={
      Array [
        "MainModule",
      ]
    }
    theme="dark"
  >
    <MenuItem
      className="menuItem"
      key="mainModule"
      onClick={[Function]}
    >
      <Icon
        type="home"
      />
      <span>
        Главное меню
      </span>
    </MenuItem>
    <MenuItem
      className="menuItem"
      key="cabinetModule"
      onClick={[Function]}
    >
      <Icon
        type="user"
      />
      <span>
        Личный кабинет
      </span>
    </MenuItem>
    <SubMenu
      className="menuItem"
      key="taskModule"
      title={
        <span>
          <Icon
            type="file-done"
          />
          <span>
            Задачи
          </span>
        </span>
      }
    >
      <MenuItem
        key="taskModule_all"
        onClick={[Function]}
      >
        Все задачи
      </MenuItem>
      <MenuItem
        key="taskModule_myTasks"
        onClick={[Function]}
      >
        Мои задачи
      </MenuItem>
      <MenuItem
        key="taskModule_сalendar"
        onClick={[Function]}
      >
        Календарь задач
      </MenuItem>
    </SubMenu>
    <SubMenu
      className="menuItem"
      key="customersModule"
      title={
        <span>
          <Icon
            type="global"
          />
          <span>
            Работа с клиентами
          </span>
        </span>
      }
    >
      <MenuItem
        key="customersModule_contacts"
        onClick={[Function]}
      >
        Контакты
      </MenuItem>
      <MenuItem
        key="customersModule_reception"
        onClick={[Function]}
      >
        Обращения
      </MenuItem>
    </SubMenu>
    <SubMenu
      className="menuItem"
      key="contactModule"
      title={
        <span>
          <Icon
            type="wechat"
          />
          <span>
            Связь
          </span>
        </span>
      }
    >
      <MenuItem
        key="contactModule_feedback"
        onClick={[Function]}
      >
        Информация
      </MenuItem>
      <MenuItem
        key="contactModule_chat"
        onClick={[Function]}
      >
        Корпоративный чат
      </MenuItem>
    </SubMenu>
    <MenuItem
      className="menuItem"
      key="statisticModule"
      onClick={[Function]}
    >
      <Icon
        type="line-chart"
      />
      <span>
        Статистика
      </span>
    </MenuItem>
    <MenuItem
      className="menuItem"
      key="settingsModule"
      onClick={[Function]}
    >
      <Icon
        type="build"
      />
      <span>
        Настройки
      </span>
    </MenuItem>
  </Menu>
</Sider>
`;
